### PR TITLE
Improve model reloading bug

### PR DESCRIPTION
**Pull request by rprospero**
There was an awkward bug where, if a plugin model was modified (so that the .py and the .pyc were out of sync), it would cause an error message until the user reloaded the models.

### CODE REVIEW

File: src/sas/sasgui/perspectives/fitting/fitpage.py [hunk 1]
```diff
@@ -1235,10 +1235,13 @@ def _on_select_model(self, event=None):
...
             # update list of plugins if new plugin is available
-            custom_model = CUSTOM_MODEL
             mod_cat = self.categorybox.GetStringSelection()
-            if mod_cat == custom_model:
+            if mod_cat == CUSTOM_MODEL:
+                temp_id = self.model.id
                 temp = self.parent.update_model_list()
+                for v in self.parent.model_dictionary.values():
+                    if v.id == temp_id:
+                        self.model = v()
```
Comment (krzywon) [inline src/sas/sasgui/perspectives/fitting/fitpage.py]: This seems like a single model is desired. A break here would exit out of the for loop as soon as it is found.

File: src/sas/sasgui/perspectives/fitting/models.py [hunk 2]
```diff
@@ -259,7 +259,9 @@ def findModels(self):
         """
         temp = {}
         if self.is_changed():
-            return  _findModels(dir)
+            temp =  _findModels(dir)
+            self.last_time_dir_modified = time.time()
```
Comment (krzywon) [inline src/sas/sasgui/perspectives/fitting/models.py]: Is this change necessary? Your changes to 311-313 should already cover the last time change.

File: src/sas/sasgui/perspectives/fitting/models.py [hunk 3]
```diff
@@ -317,13 +319,13 @@ def update(self):
...
-            for name, plug in  new_plugins.iteritems():
-                if name not in self.stored_plugins.keys():
-                    self.stored_plugins[name] = plug
-                    self.plugins.append(plug)
-                    self.model_dictionary[name] = plug
+        if new_plugins:
+            for name, plug in  new_plugins.items():
+                self.stored_plugins[name] = plug
+                self.plugins.append(plug)
+                self.model_dictionary[name] = plug
```
Comment (krzywon) [inline src/sas/sasgui/perspectives/fitting/models.py]: import CUSTOM_MODELS from fitpage.py and use it in 329.

File: src/sas/sasgui/perspectives/fitting/models.py (hunk 2 above)
Comment (rprospero) [inline src/sas/sasgui/perspectives/fitting/models.py]: Unfortunately, this change is critically necessary.  

When _findModels reads a ".py" file, it compiles it into a ".pyc" file.  This changes the last last modified time for the directory.  The actual variable, however, is not updated, so SasView was assuming that the directory had changed since its last read and was reloading again.

File: src/sas/sasgui/perspectives/fitting/models.py (hunk 3 above)
Comment (rprospero) [inline src/sas/sasgui/perspectives/fitting/models.py]: changed

File: src/sas/sasgui/perspectives/fitting/fitpage.py (hunk 1 above)
Comment (rprospero) [inline src/sas/sasgui/perspectives/fitting/fitpage.py]: changed
